flink-runtime-web/src/test/java/org/apache/flink/runtime/webmonitor/LeaderRetrievalHandlerTest.java

```diff
@@ -101,31 +87,14 @@ public void testRedirectHandler() throws Exception {
 			configuration);
 
 		try (HttpTestClient httpClient = new HttpTestClient("localhost", bootstrap.getServerPort())) {
-			// 1. without completed local address future --> Internal server error
+			// 1. no leader gateway available --> Service unavailable
 			httpClient.sendGetRequest(restPath, FutureUtils.toFiniteDuration(timeout));
 
 			HttpTestClient.SimpleHttpResponse response = httpClient.getNextResponse(FutureUtils.toFiniteDuration(timeout));
 
-			Assert.assertEquals(HttpResponseStatus.INTERNAL_SERVER_ERROR, response.getStatus());
-
-			// 2. with completed local address future but no leader gateway available --> Service unavailable
-			localAddressFuture.complete(correctAddress);
-
-			httpClient.sendGetRequest(restPath, FutureUtils.toFiniteDuration(timeout));
-
-			response = httpClient.getNextResponse(FutureUtils.toFiniteDuration(timeout));
-
 			Assert.assertEquals(HttpResponseStatus.SERVICE_UNAVAILABLE, response.getStatus());
 
-			// 3. with leader gateway which is not the one of this REST endpoints --> Redirection required
-			httpClient.sendGetRequest(restPath, FutureUtils.toFiniteDuration(timeout));
-
-			response = httpClient.getNextResponse(FutureUtils.toFiniteDuration(timeout));
-
-			Assert.assertEquals(HttpResponseStatus.TEMPORARY_REDIRECT, response.getStatus());
-			Assert.assertEquals(expectedRedirection, response.getLocation());
-
-			// 4. with local REST endpoint
+			// 2. with leader
```

Comment:
help me out here, why is the leader now suddenly available?

Reply:
with the removal of the redirection logic, there are only two cases:

1. Leader could not be retrieved yet
2. Leader has been retrieved

These two cases are tested.

Reply:
oh i get that, but this test only executes
```
httpClient.sendGetRequest(restPath, FutureUtils.toFiniteDuration(timeout));

HttpTestClient.SimpleHttpResponse response = httpClient.getNextResponse(FutureUtils.toFiniteDuration(timeout));
```
twice, with each getting a different response. What causes the response to be different? I don't see any modifications being made to the state of the rest server / futures between the 2 requests. This isn't just a timing thing, is it?

Reply:
I actually removed the mocking completely. This should make things a bit clearer.